fast64_internal/sm64/settings/properties.py

```diff
@@ -1,63 +1,161 @@
+import os
 import bpy
-from bpy.types import PropertyGroup, Scene
-from bpy.props import BoolProperty, StringProperty, EnumProperty, IntProperty, FloatProperty
+from bpy.types import PropertyGroup, UILayout, Scene, Context
+from bpy.props import BoolProperty, StringProperty, EnumProperty, IntProperty, FloatProperty, PointerProperty
+from bpy.path import abspath
 from bpy.utils import register_class, unregister_class
 
 from ...render_settings import on_update_render_settings
-from ..sm64_constants import (
-    level_enums,
-    defaultExtendSegment4,
+from ...utility import directory_path_checks, directory_ui_warnings, prop_split
+from ..sm64_constants import defaultExtendSegment4
+from ..sm64_utility import export_rom_ui_warnings, import_rom_ui_warnings
+from ..tools import SM64_AddrConvProperties
+
+from .constants import (
+    enum_refresh_versions,
+    enum_compression_formats,
+    enum_export_type,
+    enum_sm64_goal_type,
 )
-from .constants import enumRefreshVer, enumExportType, enumCompressionFormat, sm64GoalTypeEnum
 
 
-def get_legacy_export_type():
-    legacy_export_types = ("C", "Binary", "Insertable Binary")
-    scene = bpy.context.scene
-
-    for exportKey in ["animExportType", "colExportType", "DLExportType", "geoExportType"]:
-        eType = scene.pop(exportKey, None)
-        if eType is not None and legacy_export_types[eType] != "C":
-            return legacy_export_types[eType]
-
-    return "C"
+def decomp_path_update(self, context: Context):
+    directory_path_checks(abspath(self.decomp_path))
+    context.scene.fast64.settings.repo_settings_path = os.path.join(abspath(self.decomp_path), "fast64.json")
```

Comment:
I think it is possible people would want external repo settings, rather than something tied to a specific decomp repo. I would choose to update this only if the repo settings were left blank and the decomp path was updated.